partner_no_unique_bank/models/res_partner_bank.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2019 Coop IT Easy SCRL fs
+#   Houssine Bakkali <houssine@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import models
+
+
+class ResPartnerBank(models.Model):
+    _inherit = "res.partner.bank"
+
+    _sql_constraints = [
+        ("unique_number", "Check(1=1)", "Account Number must be unique!")
```

Comment:
Hi! We ended up doing something similar... but taking in account companies:

https://git.coopdevs.org/coopdevs/odoo/odoo-addons/odoo-addons/-/blob/master/remove_dup_acc_num_constraint/models/res_partner_bank.py#L7